### PR TITLE
feat(users): unifies login and registration URL generation

### LINKED ISSUE #9896: Login/register URLs
I have a feeling we need to add API to build login and registration URLs that can be filtered with a hook. Currently we hard code /login and /register, so if a site is implementing some handlers, it needs to override those. Additionally, if a site were to be an invite-only setup, there would be no way to append some invite codes to let's say an email invitation to a group. If core and plugins had a universal `elgg_get_login_url()` and `elgg_get_registration_url()`, it would be much easier to implement side-wide login/registration logic changes.

### PR DESCRIPTION
**Pull request by hypeJunction**
Adds elgg_get_registraiton_url() and elgg_get_login_url() in order to
provide a consistent way to retrieve and filter login and registration
endpoints.

Fixes #9896

### CODE REVIEW

**Comment by hypeJunction** [comment]
Any places I missed?

File: docs/guides/hooks-list.rst [hunk 1]
```diff
@@ -103,6 +103,14 @@ System hooks
 **elgg.data, page**
    Filters uncached, page-specific configuration data to pass to the client. :ref:`More info <guides/javascript#config>`
 
+**registration_url, site**
+   Filters site's registration URL. Can be used by plugins to attach invitation codes, referrer codes etc. to the registration URL.
+   ``$params`` array is volatile.
```
Comment (mrclay) [inline docs/guides/hooks-list.rst]: I don't understand

Comment (mrclay) [inline docs/guides/hooks-list.rst]: Besides this 👍

Comment (hypeJunction) [inline docs/guides/hooks-list.rst]: It depends on what the query elements are. Will make it clearer.

**Comment by hypeJunction** [comment]
@mrclay Updated the docs

**Comment by hypeJunction** [comment]
If anyone can think of other places in core that use login/register endpoints, please let me know

**Comment by jeabakker** [comment]
> If anyone can think of other places in core that use login/register endpoints, please let me know

did you check the walled garden login box?

**Comment by hypeJunction** [comment]
@jeabakker Uses core login box

**Pull request merged**